testcases/OpTestKdump.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/env python3
+# IBM_PROLOG_BEGIN_TAG
+# This is an automatically generated prolog.
+#
+# $Source: op-test-framework/testcases/OpTestKdump.py $
+#
+# OpenPOWER Automated Test Project
+#
+# Contributors Listed Below - COPYRIGHT 2017
+# [+] International Business Machines Corp.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+'''
+OpTestKdump
+------------
+This module contain testcases related to Kdump.
+TODO: Add fadump, ubuntu and LPAR support.
+'''
+import os
+import time
+import pexpect
+import unittest
+import OpTestConfiguration
+from common import OpTestHMC, OpTestFSP
+from common.OpTestError import OpTestError
+from common.OpTestSystem import OpSystemState
+from common.OpTestSSH import ConsoleState as SSHConnectionState
+from common.OpTestConstants import OpTestConstants as BMC_CONST
+from common.Exceptions import KernelOOPS, KernelKdump
+import logging
+import OpTestLogger
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+class OpTestKernelKdump(unittest.TestCase):
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+        self.cv_HOST = conf.host()
+        self.platform = conf.platform()
+        self.bmc_type = conf.args.bmc_type
+        self.util = self.cv_SYSTEM.util
+        self.console = self.cv_SYSTEM.console
+        self.stress_file = "stress-ng-0.09.58"
+        self.server_ip = "x.x.x.x"
+        self.server_pw = "xxxx"
+        self.net_path = "/var/crash_net"
+    
+    def kernel_crash(self, crash_type, dump_place):
+        '''
+        This function will test the kdump followed by system
+        reboot. it has below steps
+
+        1. Remove files under /var/crash.
+        2. Trigger kernel crash: ``echo c > /proc/sysrq-trigger``
+        '''
+        self.console.run_command("rm -rf /var/crash/*")
+        if dump_place == "net":
+            self.console.run_command('ssh root@%s "mkdir -p %s"' % (self.server_ip, self.net_path))
+            self.console.run_command('ssh root@%s "rm -rf %s/*"' % (self.server_ip, self.net_path))
+        self.console.run_command("ppc64_cpu --smt")
+        time.sleep(5)
+        self.console.run_command("echo 1 > /proc/sys/kernel/sysrq")
+        if crash_type == "manual":
+            self.console.pty.sendline("echo c > /proc/sysrq-trigger")
+            done = False
+            rc = -1
+            while not done:
```

Comment:
There needs to be a timeout that'll break out of the loop and fail the test if this takes too long. We've had enough cases of op-test getting stuck because it missed an event for some reason.